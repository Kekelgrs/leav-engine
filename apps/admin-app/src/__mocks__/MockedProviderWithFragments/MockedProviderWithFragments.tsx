// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
import {MockedProvider} from '@apollo/react-testing';
import {InMemoryCache} from 'apollo-cache-inmemory';
import React from 'react';
import {attributesFragmentMatcher} from '../fragmentMatchers/attributesFragmentMatchers';

export const getMockCacheWithFragments = () => new InMemoryCache({fragmentMatcher: attributesFragmentMatcher});

function MockedProviderWithFragments({children, ...props}) {
    // Set a new cache for each test to avoid fetching data in cache and not in provided mocks
    const mockCache = getMockCacheWithFragments();
    return (
        <MockedProvider cache={mockCache} {...props}>
            {children}
        </MockedProvider>
    );
}

export default MockedProviderWithFragments;
