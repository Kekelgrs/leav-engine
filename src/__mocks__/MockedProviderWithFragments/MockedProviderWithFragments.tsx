// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
import {InMemoryCache} from '@apollo/client';
import {MockedProvider} from '@apollo/client/testing';
import React from 'react';

// export const getMockCacheWithFragments = () => new InMemoryCache({fragmentMatcher: attributesFragmentMatcher});
export const getMockCacheWithFragments = () => new InMemoryCache();

function MockedProviderWithFragments({children, ...props}: any) {
    // Set a new cache for each test to avoid fetching data in cache and not in provided mocks
    const mockCache = getMockCacheWithFragments();
    return (
        <MockedProvider cache={mockCache} {...props}>
            {children}
        </MockedProvider>
    );
}

export default MockedProviderWithFragments;
