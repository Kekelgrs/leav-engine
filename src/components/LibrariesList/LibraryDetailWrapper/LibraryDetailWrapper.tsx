import React from 'react';
import {useHistory, useParams} from 'react-router-dom';
import {Button} from 'semantic-ui-react';
import LibraryDetail from '../LibraryDetail/LibraryDetail';

function LibraryDetailWrapper(): JSX.Element {
    const {libId, libQueryName} = useParams();
    const history = useHistory();

    const goBack = () => {
        history.goBack();
    };

    return (
        <>
            <Button icon="arrow left" onClick={goBack} />
            <LibraryDetail libId={libId ?? ''} libQueryName={libQueryName ?? ''} />
        </>
    );
}

export default LibraryDetailWrapper;
