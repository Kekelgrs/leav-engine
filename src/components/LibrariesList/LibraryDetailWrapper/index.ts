import LibraryDetailWrapper from './LibraryDetailWrapper';
export default LibraryDetailWrapper;
