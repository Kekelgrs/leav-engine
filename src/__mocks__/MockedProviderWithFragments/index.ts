import MockedProviderWithFragments from './MockedProviderWithFragments';
export default MockedProviderWithFragments;
