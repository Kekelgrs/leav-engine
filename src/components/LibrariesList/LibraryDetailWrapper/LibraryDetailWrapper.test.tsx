import {render} from 'enzyme';
import React from 'react';
import MockedProviderWithFragments from '../../../__mocks__/MockedProviderWithFragments';
import LibraryDetailWrapper from './LibraryDetailWrapper';

jest.mock('react-router-dom', () => ({
    useParams: jest.fn(() => ({libId: 'test', libQueryName: 'test'})),
    useHistory: jest.fn
}));

describe('LibraryDetailWrapper', () => {
    test('Snapshot test', async () => {
        const comp = render(
            <MockedProviderWithFragments>
                <LibraryDetailWrapper />
            </MockedProviderWithFragments>
        );

        expect(comp).toMatchSnapshot();
    });
});
