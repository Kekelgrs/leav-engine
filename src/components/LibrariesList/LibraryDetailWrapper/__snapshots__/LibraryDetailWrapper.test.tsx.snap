// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LibraryDetailWrapper Snapshot test 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "ui icon button",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "aria-hidden": "true",
          "class": "arrow left icon",
        },
        "children": Array [],
        "name": "i",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "aria-hidden": undefined,
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "aria-hidden": undefined,
          "class": undefined,
        },
      },
    ],
    "name": "button",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": Object {
      "attribs": Object {},
      "children": Array [
        Object {
          "data": "loading",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "text",
        },
      ],
      "name": "div",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": [Circular],
      "root": Object {
        "attribs": Object {},
        "children": Array [
          [Circular],
          [Circular],
        ],
        "name": "root",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": null,
        "prev": null,
        "type": "root",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      "type": "tag",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
        Object {
          "attribs": Object {},
          "children": Array [
            Object {
              "data": "loading",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": null,
          "prev": [Circular],
          "root": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "1": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "data": "loading",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": Object {
      "attribs": Object {
        "class": "ui icon button",
      },
      "children": Array [
        Object {
          "attribs": Object {
            "aria-hidden": "true",
            "class": "arrow left icon",
          },
          "children": Array [],
          "name": "i",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "aria-hidden": undefined,
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "aria-hidden": undefined,
            "class": undefined,
          },
        },
      ],
      "name": "button",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": [Circular],
      "parent": null,
      "prev": null,
      "root": Object {
        "attribs": Object {},
        "children": Array [
          [Circular],
          [Circular],
        ],
        "name": "root",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": null,
        "prev": null,
        "type": "root",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      "type": "tag",
      "x-attribsNamespace": Object {
        "class": undefined,
      },
      "x-attribsPrefix": Object {
        "class": undefined,
      },
    },
    "root": Object {
      "attribs": Object {},
      "children": Array [
        Object {
          "attribs": Object {
            "class": "ui icon button",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "aria-hidden": "true",
                "class": "arrow left icon",
              },
              "children": Array [],
              "name": "i",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "aria-hidden": undefined,
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "aria-hidden": undefined,
                "class": undefined,
              },
            },
          ],
          "name": "button",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": null,
          "prev": null,
          "root": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "_root": [Circular],
  "length": 2,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
